src/backend/cdb/cdbpath.c

```diff
@@ -136,6 +138,22 @@ cdbpath_create_motion_path(PlannerInfo *root,
 			return subpath;
 		}
 
+		/* No motion needed if subpath can run anywhere giving same output. */
```

Comment:
I suppose there is no need to move `if (CdbPathLocus_IsGeneral(subpath->locus))` code block. It seems that we only need to move up
```c
		/* Fail if caller refuses motion. */
		if (require_existing_order &&
			!pathkeys)
			return NULL;
```

Reply:
Do you think `require_existing_order` is false or `pathkeys` is not NULL, when locus type is General?

Reply:
I was wrong =)